### ISSUE TITLE
Schema Versioning

### ISSUE THREAD

**Issue by sunnya97** [cosmos/chain-registry#142]
How can we do versioning for the schemas so that we know when there's breaking changes?

**Comment by hxrts** [comment]
This is a very elegant schema versioning system but it's likely far too much work to actually implement: https://www.inkandswitch.com/cambria/

I think we'd want to do something that lines up with a standard format like json-ld or DID, may want to move to self-describing data format.

**Comment by JeremyParish69** [comment]
Will be wanting to use chain reg as a sub-module, so this seems increasingly important.

**Comment by hxrts** [comment]
> Will be wanting to use chain reg as a sub-module, so this seems increasingly important.

Interesting. Can you say more about your use-case? Like a git submodule?

**Comment by JeremyParish69** [comment]
yes, @hxrts , as a git sub-module to be used as a reference for chain and asset information, IBC channel info, token images, etc., and remove all local hard-coded information about tokens and chains. Any changes about a token, e.g., changing symbol of LUNA to LUNC, desired to take effect on Osmosis would instead be done on the chain registry, and anything we need to construct will be done so through lookups using only ( chain_name + coinMinimalDenom ) as primary keys, which are assumed to be immutable.
